pkg/iptables/iptables.go

```diff
@@ -4,25 +4,86 @@
+const redirectLocalRule = "OUTPUT " + // For local traffic
+	"-t nat " + // Traversing the nat table
+	"-s 127.0.0.0/8 -d 127.0.0.0/8 " + // Coming from and directed to the loopback address, i.e. not the pod IP.
+	"-p tcp --dport %d " + // Sent to the upstream application's port
+	"-j REDIRECT --to-port %d" // Forward it to the proxy address
+
+// redirectExternalRule is a netfilter rule that intercepts external traffic directed to the application and redirects
+// it to the proxy.
+// Traffic created by the proxy itself to the application traverses is not redirected by this rule as it traverses the
+// OUTPUT chain, not PREROUTING.
+const redirectExternalRule = "PREROUTING " + // For remote traffic
+	"-t nat " + // Traversing the nat table
+	"! -i lo " + // Not coming form loopback. This is technically not needed, but doesn't hurt and helps readability.
+	"-p tcp --dport %d " + // Sent to the upstream application's port
+	"-j REDIRECT --to-port %d" // Forward it to the proxy address
+
+// resetLocalRule is a netfilter rule that resets established connections (i.e. that have not been redirected) coming
+// to and from the loopback address.
+// This rule matches connections from sidecars and `kubectl port-forward`.
+// Connections from the proxy itself do not match this rule, as they are directed to the pod's external IP and not
+// loopback.
+const resetLocalRule = "INPUT " + // For traffic traversing the INPUT chain
+	"-i lo " + // On the loopback interface
+	"-s 127.0.0.0/8 -d 127.0.0.0/8 " + // Coming from and directed to the loopback address
+	"-p tcp --dport %d " + // Directed to the upstream application's port
+	"-m state --state ESTABLISHED " + // That are already ESTABLISHED, i.e. not before they are redirected
+	"-j REJECT --reject-with tcp-reset" // Reject it
+
+// resetExternalRule is a netfilter rule that resets established connections (i.e. that have not been redirected) coming
+// from anywhere except the local IP.
+// This rule matches external connections to the pod's IP address.
+// Connections from the proxy itself do not match this rule, as they are originated from the loopback address.
+const resetExternalRule = "INPUT " + // For traffic traversing the INPUT chain
+	"! -i lo " + // Not coming form loopback. This is technically not needed, but doesn't hurt and helps readability.
+	"-p tcp --dport %d " + // Directed to the upstream application's port
+	"-m state --state ESTABLISHED " + // That are already ESTABLISHED, i.e. not before they are redirected
+	"-j REJECT --reject-with tcp-reset" // Reject it
 
 // TrafficRedirectionSpec specifies the redirection of traffic to a destination
 type TrafficRedirectionSpec struct {
-	// Interface on which the traffic will be intercepted
-	Iface string
-	// Destination port of the traffic to be redirected
-	DestinationPort uint
-	// Port the traffic will be redirected to
-	RedirectPort uint
+	// ProxyPort is the port where the proxy is listening at.
+	ProxyPort uint
```

Comment:
I juggled a bit with the names because I was confused about the originals. `Destination` and `Redirect` would be interpreted in both ways, e.g. Is destination the "new" destination, or the destination for the rule? Is Redirect the port where the traffic is redirected, or the port where we set up the redirection?

It's true that the Proxy may not be the best choice here. Let me think a bit about it and see if I can think about something else.

Reply:
What about `FromPort` and `ToPort`? I think it makes sense in the context of something that redirects: Redirect from somewhere to somewhere.

Reply:
I'm sure I can't beat that.

Reply:
Gave this a try in 59345a41b7bdb76eb1a32b1890d4ddc917869b18. I think now names are more consistent, but feels a bit awkward when setting it up:

```go
tr := &iptables.TrafficRedirectionSpec{
	FromPort: targetPort, // Redirect traffic from the application (target) port...
	ToPort:   port,       // to the proxy port.
}
```

Not sure if we should prefer the "stay true to the use case" rule here.

Reply:
Yes, I agree. That looks awkward, and maybe the only solution is to rename the port-related arguments all along the path from the API down the redirector, which is definitely outside the scope of the PR.

As it seems we are in agreement here and this is a minor point not particularly relevant to this PR, so I suggest keeping the original naming because it at least makes reference to the iptables command, so is consistent with the scope of the module:

`A PREROUTING  --dport <destination port> -j REDIRECT --to-port  <redirect port>`

If anything, we can change it to the (maybe too verbose) `redirect-to-port`

Reply:
I'm not sure if my previous comment was clear enough: by "original naming" I meant the one before this PR.

Reply:
> maybe the only solution is to rename the port-related arguments all along the path from the API down the redirector, which is definitely outside the scope of the PR.

Agreed with both, after the changes we've done lately we could benefit from a renaming ensuring consistency along the call stack. Reverted to its original names, which this PR is definitely no the place to do. Reverted in https://github.com/grafana/xk6-disruptor/pull/231/commits/d41c47ee3cfb6dde1566ab977edb74696b502f70.

Reply:
At this point I've given up with `fixup!` commits, I'll manually squash things together once the PR gets approved.